pkg/controller/utils/component.go

```diff
@@ -71,16 +71,20 @@ func (c componentHandler) CreateOrUpdate(ctx context.Context, component render.C
 	daemonSets := []types.NamespacedName{}
 	deployments := []types.NamespacedName{}
 	statefulsets := []types.NamespacedName{}
-	for _, obj := range component.Objects() {
+	objsToCreate, objsToDelete := component.Objects()
+
+	var err error
```

Comment:
I don't think you need to move these outside here. You can probably actually just delete line 82, since the key is only used within that loop IIUC.

Reply:
It's used on line 163 too

Reply:
Right, but we could just do a `:=` there as well and then we don't need to declare these up here at all, right?

Reply:
Sure.. this works as well, right?

Reply:
It totally works, it's just a slightly confusing / non-standard structure.  Declaring variables up-front like this usually implies that the value assigned to the variable in some earlier context will be used in a later on, but that's not what we're doing here.

I'm just suggesting the change so that future readers don't think that there's some special link between the usage of key in the first loop and the usage in the second.

Reply:
Fair enough